art/metrics/metrics.py

```diff
@@ -352,3 +352,44 @@ def clever_t(classifier, x, target_class, nb_batches, batch_size, radius, norm,
     score = np.min([-value[0] / loc, radius])
 
     return score
+
+
+def wasserstein_distance(u_values, v_values, u_weights=None, v_weights=None):
+    """
+    Compute the first Wasserstein distance between two 1D distributions.
+
+    :param u_values: Values of first distribution with shape (nb_samples, feature_dim_1, ..., feature_dim_n)
+    :type u_values: `np.ndarray`
+    :param v_values: Values of second distribution with shape (nb_samples, feature_dim_1, ..., feature_dim_n)
+    :type v_values: `np.ndarray`
+    :param u_weights: Weight for each value. If None equal weights will be used.
+    :type u_weights: `np.ndarray`
+    :param v_weights: Weight for each value. If None equal weights will be used.
+    :type v_weights: `np.ndarray`
+    :return: The Wasserstein distance between the two distributions
+    :rtype: `np.ndarray`
+    """
+    from scipy.stats import wasserstein_distance
+
+    assert u_values.shape == v_values.shape
+    if u_weights is not None:
+        assert v_weights is not None
+    if u_weights is None:
+        assert v_weights is None
+    if u_weights is not None and v_weights is not None:
+        assert u_weights.shape == v_weights.shape
+    if u_weights is not None:
+        assert u_values.shape[0] == u_weights.shape[0]
```

Comment:
It seems to me that you can place the last two asserts in the first condition block and the second `if` can be replaced with an `else`.
```suggestion
    if u_weights is not None:
        assert v_weights is not None
        assert u_weights.shape == v_weights.shape
        assert u_values.shape[0] == u_weights.shape[0]
    else:
        assert v_weights is None
```

Reply:
Nice!